### DISCUSSION TITLE
Milvus Cluster mode with shared components

### DISCUSSION THREAD

**Discussion by nairan-deshaw** [milvus-io/milvus#35685] [Q&A and General discussion]
Hi, we are trying evaluate the feasibility of a Milvus farm within Kubernetes where we can share components like minio, etcd, pulsar etc. and have multiple instances of milvus in separate namespaces. Is this supported with the current design of Milvus should we tweak the configurations or is there a one to one mapping between each milvus database and the surrounding components it uses.

**Comment by xiaofan-luan** [comment]
yes we do.

There are two way to do so.

1. Share all dependencies, build a cluster with different storage path/ S3 bucket name
2. Build one cluster, use resource group to isolate the traffic

**Comment by xiaofan-luan** [comment]
Also, please be aware of using zilliz cloud might be the easiest way to do so

**Comment by nairan-deshaw** [comment]
Thanks for the reply @xiaofan-luan, can you please elaborate on the two approaches, how do we go about configuring milvus to use different paths and what is resource group isolation?

**Reply by xiaofan-luan** [reply]
there is some config named root_path for both S3 an minio. For kafka/pulsar there is some other config to control topics.

More details about resource groups https://milvus.io/docs/resource_group.md
